cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -759,70 +681,64 @@ private void assertNonNullParent(EntityId parent, Authorizable.AuthorizableType
   }
 
   /**
-   * Maps {@link EntityId} to a {@link List} of {@link Authorizable}
+   * Maps {@link co.cask.cdap.proto.security.Authorizable} to a {@link List} of {@link Authorizable}
    * by recursively working its way from a given entity.
    *
-   * @param entityId {@link EntityId} the entity which needs to be mapped to a list of authorizables
-   * @param authorizables {@link List} of {@link Authorizable} to
-   * add authorizables to
+   * @param curType entity type
+   * @param authorizable the cdap authorizable which needs to be mapped
+   * @param sentryAuthorizables list of sentry authorizable
    */
-  void toAuthorizables(EntityId entityId, List<? super Authorizable> authorizables) {
-    EntityType entityType = entityId.getEntityType();
-    switch (entityType) {
+  void toAuthorizables(EntityType curType, co.cask.cdap.proto.security.Authorizable authorizable,
+                       List<? super Authorizable> sentryAuthorizables) {
+    switch (curType) {
       case INSTANCE:
-        authorizables.add(new Instance(((InstanceId) entityId).getInstance()));
+        sentryAuthorizables.add(new Instance(authorizable.getEntityParts().get(EntityType.INSTANCE)));
```

Comment:
It can't be since there is no instance in the map as cdap entities does not have instance in hierarchy.